### PR TITLE
Modal window editing

### PR DESCRIPTION
**Pull request by VitaliyTheKeyboardist**

### CODE REVIEW

File: src/app/(home)/page.tsx [hunk 1]
```diff
@@ -1,3 +1,5 @@
+"use client";
+
```
Comment (AlexMakovetskiy) [inline src/app/(home)/page.tsx:2]: this line is unnecessary

File: src/pages/homeLayout/cookiePopupWindow/cookiePopupWindow.tsx
Comment (AlexMakovetskiy) [inline src/pages/homeLayout/cookiePopupWindow/cookiePopupWindow.tsx:1]: can you remane this folder and it's files from "CookiePopupWindow" to "CookiePopupWindowPage"?

File: src/pages/homeLayout/cookiePopupWindow/cookiePopupWindow.tsx [hunk 2]
```diff
@@ -1,29 +1,38 @@
...
 
-import closeIcon from "../../../assets/pages/mainPage/closeIcon.svg";
+import useAppSelector from "../../../hooks/useAppSelector";
+import cookieStatusSelector from "../../../services/redux/features/cookieStatus/cookieStatusSelector";
+import useAppDispatch from "../../../hooks/useAppDispatch";
+import { setCookieStatus } from "../../../services/redux/features/cookieStatus/cookieStatusSlice";
 
 import styles from "./cookiePopupWindow.module.scss";
 
 const CookiePopupWindow = () => {
```
Comment (AlexMakovetskiy) [inline src/pages/homeLayout/cookiePopupWindow/cookiePopupWindow.tsx:17]: can you remane this page component from "CookiePopupWindow" to "CookiePopupWindowPage"?

File: src/pages/homeLayout/cookiePopupWindow/cookiePopupWindow.tsx [hunk 3]
```diff
@@ -1,29 +1,38 @@
...
+		setIsOpen(false);
+		dispatch(setCookieStatus("rejected"));
+	};
+
 	return (
-		<div className={cn(styles.popupContainer, { [styles.popupContainer_hidden]: !isOpen })}>
-			<div className={styles.textContentBlock}>
+		<div
+			className={cn(styles.popupContainer, { [styles.popupContainer_hidden]: status === "confirmed" || !isOpen })}
+			onClick={handleClick}>
```
Comment (AlexMakovetskiy) [inline src/pages/homeLayout/cookiePopupWindow/cookiePopupWindow.tsx:32]: add role="button" attribute for this div element

File: src/pages/homeLayout/cookiePopupWindow/cookiePopupWindow.tsx [hunk 4]
```diff
@@ -1,29 +1,38 @@
...
+			<div className={styles.textContentBlock} onClick={(e) => e.stopPropagation()}>
```
Comment (AlexMakovetskiy) [inline src/pages/homeLayout/cookiePopupWindow/cookiePopupWindow.tsx:33]: add the same attribute for this div element

File: src/services/redux/index.ts [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
+// import sliceReducer from "./features/userData/UserDataSlice";
 import dataReducer from "./features/infoPart/InfoPartSlice";
+import statusReducer from "./features/cookieStatus/cookieStatusSlice";
 
 const store = configureStore({
 	reducer: {
-		user: sliceReducer,
+		// user: sliceReducer,
 		data: dataReducer,
+		status: statusReducer,
```
Comment (AlexMakovetskiy) [inline src/services/redux/index.ts:11]: better remove unnecessary comment lines

File: src/services/redux/features/cookieStatus/cookieStatusSlice.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+const initialState: ICookieStatus = {
+	status: "pending",
+};
+
+const cookieStatus = createSlice({
+	name: "cookieStatus",
+	initialState,
+	reducers: {
+		setCookieStatus(state, action) {
```
Comment (AlexMakovetskiy) [inline src/services/redux/features/cookieStatus/cookieStatusSlice.ts:13]: state: ICookieStatus, action: PayloadAction<CookieStatusList>
better add types for reducer

File: src/app/(home)/page.tsx (hunk 1 above)
Comment (VitaliyTheKeyboardist) [reply src/app/(home)/page.tsx:2]: here is the problem: "You're importing a component that needs useState. It only works in a Client Component but none of its parents are marked with "use client", so they're Server Components by default."

File: src/pages/homeLayout/cookiePopupWindow/cookiePopupWindow.tsx (hunk 2 above)
Comment (VitaliyTheKeyboardist) [reply src/pages/homeLayout/cookiePopupWindow/cookiePopupWindow.tsx:17]: what do you mean here?

Comment (AlexMakovetskiy) [reply src/pages/homeLayout/cookiePopupWindow/cookiePopupWindow.tsx]: const CookiePopupWindow = () => {
=>
const CookiePopupWindowPage= () => {
just rename page component

File: src/app/(home)/page.tsx (hunk 1 above)
Comment (AlexMakovetskiy) [reply src/app/(home)/page.tsx:2]: all files you import in this file don't contain the line "use client" and don't use useState hook

Comment (AlexMakovetskiy) [reply src/app/(home)/page.tsx:2]: maybe you meant this file?
src/app/(home)/layout.tsx

Comment (VitaliyTheKeyboardist) [reply src/app/(home)/page.tsx:2]: *facepalm* I'm sorry

File: src/pages/homeLayout/cookiePopupWindow/cookiePopupWindow.tsx (hunk 2 above)
Comment (VitaliyTheKeyboardist) [reply src/pages/homeLayout/cookiePopupWindow/cookiePopupWindow.tsx]: it's already done. It seemed like the same conversation twice

**Review by AlexMakovetskiy — APPROVED** [review]

**Pull request merged**